### PR TITLE
Update splunklogging SplunkHandler class with new grains on refresh

### PR DESCRIPTION
**Pull request by mew1033**
Also remove unneeded code

### CODE REVIEW

**Comment by fossam** [comment]
This pr seems to be making subtle changes to already subtle code. Dealing with the downstream effects of hard to foresee corner cases has been an ongoing issue. I am afraid that it will be a bigger issue going forward unless we make it easier to understand the changes to those who will need to look at this later.

@mew1033, can you record an explanation of these changes in two parts:
* at the high level the problem the pr aims to solve
* at the low level how these particular changes fix or avoid issues (preferably without affecting anything else).

File: hubblestack/splunklogging.py [hunk 1]
```diff
@@ -171,6 +137,14 @@ def emit(self, record):
             hec.flushBatch()
         return True
 
+    def update_event_std_info(self):
+        """
+        Update the `event` template in the `endpoint_list` object. This allows
+        grains and other values that were updated to be updated here.
+        """
+        for entry in self.endpoint_list:
+            entry[1] = entry[1].update(hubblestack.utils.stdrec.std_info())
```
Comment (jettero) [inline hubblestack/splunklogging.py:146]: This should probably be `entry[1].update(...)` the return value of `dict.update` is None.

**Review by jettero — CHANGES_REQUESTED** [review]

File: hubblestack/splunklogging.py (hunk 1 above)
Comment (mew1033) [reply hubblestack/splunklogging.py]: Fixed. Thanks!

**Comment by mew1033** [comment]
@fossam I agree with you. This is a confusing change, and I want to do it better. However, we need this in 3.0 and 4.0. Once we fixed that, we should have a bigger discussion about doing this the right way.

**Comment by mew1033** [comment]
I created #933 to get this change merged into 4.0. We can continue to use this PR to discuss the proper way to implement this/revamp the logging.

**Review by jettero — APPROVED** [review]

**Pull request merged**